pontoon/base/models.py

```diff
@@ -2330,6 +2358,58 @@ def cleaned_key(self):
     def __str__(self):
         return self.string
 
+    def stats_states_map(self, locale):
+        translations = list(
+            self.translation_set
+                .filter(locale=locale)
+                .prefetch_related('errors', 'warnings')
+        )
+
+        approved_strings_count = len([
+            t for t in translations
+            if t.approved
```

Comment:
We need to change this line to:
```python
if t.approved and not t.errors.exists() and not t.warnings.exists()
```
If you check `calculate_stats` and the `translated()` filter, you'll see we do the same thing there.